File: src/test/cpp/movelisttest.cpp
/*
 * Copyright (C) 2013-2015 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

#include "movelist.h"

#include "gtest/gtest.h"

using namespace pulse;

TEST(movelisttest, test) {
  MoveList<MoveEntry> moveList;

  EXPECT_EQ(0, moveList.size);

  moveList.entries[moveList.size++]->move = 1;
  EXPECT_EQ(1, moveList.size);
}

File: pulse-java/src/test/java/com/fluxchess/pulse/MoveListTest.java
/*
 * Copyright 2013-2023 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */
package com.fluxchess.pulse;

import org.junit.jupiter.api.Test;

import static com.fluxchess.pulse.MoveList.MoveEntry;
import static org.assertj.core.api.Assertions.assertThat;

class MoveListTest {

	@Test
	void test() {
		MoveList<MoveEntry> moveList = new MoveList<>(MoveEntry.class);

		assertThat(moveList.size).isEqualTo(0);

		moveList.entries[moveList.size++].move = 1;
		assertThat(moveList.size).isEqualTo(1);
	}
}
